apis/network/v1alpha3/types.go

```diff
@@ -218,10 +218,198 @@ type Subnet struct {
+	metav1.TypeMeta   `json:",inline"`
+	metav1.ObjectMeta `json:"metadata,omitempty"`
+
+	// SecurityRulePropertiesFormat - Properties of the security rule.
+	Properties SecurityRulePropertiesFormat `json:"properties,omitempty"`
+	// Name - The name of the resource that is unique within a resource group. This name can be used to access the resource.
+	Name string `json:"name,omitempty"`
+	// Etag - A unique read-only string that changes whenever the resource is updated.
+	Etag string `json:"etag,omitempty"`
+	// ID - Resource ID.
+	ID string `json:"id,omitempty"`
+}
+
+// A SecurityGroupSpec defines the desired state of a SecurityGroup.
+type SecurityGroupSpec struct {
+	runtimev1alpha1.ResourceSpec `json:",inline"`
+
+	// ResourceGroupName - Name of the SecurityGroup's resource group.
+	ResourceGroupName string `json:"resourceGroupName,omitempty"`
+
+	// ResourceGroupNameRef - A reference to the the SecurityGroup's resource
+	// group.
+	ResourceGroupNameRef *runtimev1alpha1.Reference `json:"resourceGroupNameRef,omitempty"`
+
+	// ResourceGroupNameSelector - Select a reference to the the Security
+	// group's resource group.
+	ResourceGroupNameSelector *runtimev1alpha1.Selector `json:"resourceGroupNameSelector,omitempty"`
+
+	// Location - Resource location.
+	Location string `json:"location"`
+
+	//SecurityGroPropertiesFormat - Properties of security group
+	SecurityGroupPropertiesFormat `json:"properties,omitempty"`
+
+	// Tags - Resource tags.
+	// +optional
+	Tags map[string]string `json:"tags,omitempty"`
+}
+
+// A SecurityGroupStatus represents the observed status of a SecurityGroup.
+type SecurityGroupStatus struct {
+	runtimev1alpha1.ResourceStatus `json:",inline"`
+
+	// State of this SecurityGroup.
+	State string `json:"state,omitempty"`
+
+	// A Message providing detail about the state of this SecurityGroup, if
+	// any.
+	Message string `json:"message,omitempty"`
+
+	// ID of this SecurityGroup.
+	ID string `json:"id,omitempty"`
+
+	// Etag - A unique read-only string that changes whenever the resource is
+	// updated.
+	Etag string `json:"etag,omitempty"`
+
+	// ResourceGUID - The GUID of this SecurityGroup.
+	ResourceGUID string `json:"resourceGuid,omitempty"`
+
+	// Type of this SecurityGroup.
+	Type string `json:"type,omitempty"`
+}
+
+// SecurityGroupPropertiesFormat network Security Group resource.
+type SecurityGroupPropertiesFormat struct {
+	// SecurityRules - A collection of security rules of the network security group.
+	SecurityRules *[]SecurityRule `json:"securityRules,omitempty"`
+	// DefaultSecurityRules - The default security rules of network security group.
+	DefaultSecurityRules *[]SecurityRule `json:"defaultSecurityRules,omitempty"`
+	// NetworkInterfaces - READ-ONLY; A collection of references to network interfaces.
+	//NetworkInterfaces *[]Interface `json:"networkInterfaces,omitempty"`
+	// Subnets - READ-ONLY; A collection of references to subnets.
+	//Subnets *[]Subnet `json:"subnets,omitempty"`
```

Comment:
Should these be implemented?

Reply:
As of now we are not using it so we are removing it as of now, in future we could have a plan to implement this.